service/controller/resource/values/create.go

```diff
@@ -0,0 +1,68 @@
+package values
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/giantswarm/config-controller/pkg/generator/key"
+	controllerkey "github.com/giantswarm/config-controller/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	app, err := controllerkey.ToAppCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configVersion, ok := app.GetAnnotations()[key.ConfigVersion]
+	if !ok {
+		r.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("App CR %q is missing %q annotation", app.Name, key.ConfigVersion))
```

Comment:
This is just a convention we have:

```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("App CR %q is missing %q annotation", app.Name, key.ConfigVersion))
		r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling resource")
```